catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePollerRunner.java

```diff
@@ -198,4 +186,76 @@ public void shutdown() {
         }
         LOGGER.trace("Status threads shut down");
     }
+
+    private SourceKey getSourceKey(Source source) {
+        return new SourceKey(source);
+    }
+
+    private static class SourceKey {
```

Comment:
Agree we should have `equals()` and `hashcode()`. Can get by without the `Comparable` interface because the `Comparator` on the map using it is explicitly defined.